test/Riok.Mapperly.Tests/Mapping/EnumerableTest.cs

```diff
@@ -10,7 +10,7 @@ public void ArrayToArrayOfPrimitiveTypes()
             "int[]");
         TestHelper.GenerateSingleMapperMethodBody(source)
             .Should()
-            .Be("return System.Linq.Enumerable.ToArray(source);");
+            .Be("return (int[])source.Clone();");
```

Comment:
Could you add tests for strings (where Array.Clone()  should be used) and a non-primitive type (where Clone() shouldn't be used)?

Reply:
Added tests.